EventGenerator/src/SimpleAntiprotonGun_module.cc

```diff
@@ -0,0 +1,215 @@
+// Generate antiproton events upstream of the stopping target using rough approximations
+// Michael MacKenize, 2024
+
+
+#include <iostream>
+#include <string>
+#include <cmath>
+
+#include "CLHEP/Vector/ThreeVector.h"
+#include "CLHEP/Vector/LorentzVector.h"
+#include "CLHEP/Random/RandomEngine.h"
+#include "CLHEP/Random/RandExponential.h"
+#include "CLHEP/Random/RandFlat.h"
+#include "CLHEP/Units/PhysicalConstants.h"
+
+#include "art/Framework/Core/EDProducer.h"
+#include "art/Framework/Principal/Event.h"
+#include "art/Framework/Principal/Run.h"
+#include "art/Framework/Principal/Handle.h"
+#include "art/Framework/Services/Registry/ServiceHandle.h"
+#include "art_root_io/TFileService.h"
+
+#include "Offline/ConfigTools/inc/ConfigFileLookupPolicy.hh"
+#include "Offline/SeedService/inc/SeedService.hh"
+#include "Offline/GlobalConstantsService/inc/GlobalConstantsHandle.hh"
+#include "Offline/GlobalConstantsService/inc/ParticleDataList.hh"
+#include "Offline/GlobalConstantsService/inc/PhysicsParams.hh"
+#include "Offline/DataProducts/inc/PDGCode.hh"
+#include "Offline/MCDataProducts/inc/GenId.hh"
+#include "Offline/MCDataProducts/inc/GenParticle.hh"
+#include "Offline/GeometryService/inc/GeomHandle.hh"
+#include "Offline/StoppingTargetGeom/inc/StoppingTarget.hh"
+#include "Offline/StoppingTargetGeom/inc/TargetFoil.hh"
+
+#include "TH1.h"
+
+namespace mu2e {
+
+  //================================================================
+  class SimpleAntiprotonGun : public art::EDProducer {
+    struct Config {
+      using Name=fhicl::Name;
+      using Comment=fhicl::Comment;
+      fhicl::Atom<double> timeArrivalRate{Name("timeArrivalRate"), Comment("Exponential rate for antiproton arrival"), 130.};
+      fhicl::Atom<double> t0{Name("t0"), Comment("Start time for the antiproton distribution"), 600.}; //time distribution falls again below 600 ns
```

Comment:
It looks like this gun fires AP into the stopping target 'point blank'. I was rather expecting it would fire from the production target, to allow the beamline to naturally provide timing transport and momentum selection. In that case, the initial timing distribution could be the primary proton timing, and the dependence on the initially assumed momentum distribution would be muted.

Reply:
in the antiproton studies for SU2020 we have found that Geant4 overestimates the probability to emit in the backward direction the antiprotons produced by proton interactions in the stopping target. A patch using  a correction obtained comparing Geant4 and MCNP has been introduced using event weights. The generation at the production target is also important to produced the delayed RPC produced by antiproton annihilation in the TS, these give a non negligible contribution to the antiproton related background. I think the goal of Michael's generator is limited to the study of antiproton annihilations in the production target. If it's just a tool to test the multiple track reconstruction the distribution of the anitproton annihilation position and time should not be so relevant. Perhaps a flat distribution could be more easily reweighted with the wanted distribution in case we try for example to change the production target

Reply:
The idea here was to make a simple sample of annihilation events in the stopping target for tracking studies. I believe the survival of the antiprotons are very low along the beamline, so resampling stages were originally setup along the path (I believe before the TS, before the TS window, etc.). This workflow would need to be implemented and debugged by an expert in order to produce a useful antiproton dataset at the ST. The goal of this generator was to approximate the final result at the ST. I can update the arrival time to be flat, so studies can be performed easily be reweighting the time and momentum distributions. Overall, the goal here was to have a quick workflow for annihilations until someone more expert implements a proper workflow and produces a large dataset of stopping target annihilations.